FWCore/Framework/src/EventProcessor.cc

```diff
@@ -2203,64 +2205,87 @@ namespace edm {
   }
 
   void EventProcessor::handleNextEventForStreamAsync(WaitingTaskHolder iTask, unsigned int iStreamIndex) {
+    auto handleNextEventStatus =
+        std::make_shared<std::atomic<unsigned int>>(static_cast<unsigned int>(HandleNextEventStatus::kRunning));
```

Comment:
How about
```suggestion
        std::make_shared<std::atomic<HandleNextEventStatus>>(static_cast<unsigned int>(HandleNextEventStatus::kRunning));
```
? On a quick test an atomic of `enum class` would be lock free on gcc 12.

Reply:
This becomes moot if I change to follow Chris's suggestion.

Does an ```atomic<EnumType>``` work and perform as well as an ```atomic<unsigned int>```? I know the second one works. I wasn't sure about the first and my google searches so far have failed to answer the question. I know there is a specialization for unsigned int but maybe the instantiation for an enum is just as good.

I'll change this if I don't delete it entirely. Getting rid of the casts would be an improvement.

Reply:
As far as I've understood, atomic of enum works as an atomic of the underlying integer. You could specify the underlying integer type explicitly in the `enum class` definition.

Alternatively I'd use old `enum` instead of `enum class`, because the enumeration is always used in an integer context (and therefore, to me, the type casts only add noise).